capstone_project/src/test/java/com/google/sps/PostFilterTest.java

```diff
@@ -0,0 +1,123 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import static org.junit.Assert.assertEquals;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Text;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.gson.Gson;
+import com.google.sps.data.Post;
+import com.google.sps.data.PostFilter;
+import com.google.sps.data.PostService;
+import com.google.sps.servlets.FilterPostsServlet;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+
+@RunWith(JUnit4.class)
+public final class PostFilterTest extends Mockito {
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private HttpServletRequest request;
+  private HttpServletResponse response;
+  private LocalServiceTestHelper serviceHelper =
+      new LocalServiceTestHelper(
+              new LocalDatastoreServiceTestConfig(), new LocalUserServiceTestConfig())
+          .setEnvIsLoggedIn(true);
+
+  private final String SEARCH_QUERY = "Test University";
+  private final String DEFAULT_STRING = "Some Text";
+  private Entity post1;
```

Comment:
Not necessary, but it might be helpful to comment either here, or in the set up, a very short description about each test post, or to briefly explain the expected values in comments before each test. Reading the set up is somewhat difficult just because there is a lot of text.

Reply:
Done